src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs

```diff
@@ -114,87 +112,67 @@ private async Task ProbeCluster(ClusterState cluster)
-                    {
-                        edi = ExceptionDispatchInfo.Capture(ex);
-                        Log.DestinationProbingFailed(_logger, allDestinations[i].DestinationId, cluster.ClusterId, ex);
-                    }
-                    probingResults[i] = new DestinationProbingResult(allDestinations[i], response, edi?.SourceException);
-                }
+            for (var i = 0; i < probeResults.Length; i++)
+            {
+                probeResults[i] = await probeTasks[i];
+            }
 
-                policy.ProbingCompleted(cluster, probingResults);
+            try
+            {
+                policy.ProbingCompleted(cluster, probeResults);
             }
             catch (Exception ex)
             {
                 Log.ActiveHealthProbingFailedOnCluster(_logger, cluster.ClusterId, ex);
             }
             finally
             {
-                foreach (var probeTask in probeTasks)
-                {
-                    try
-                    {
-                        try
-                        {
-                            probeTask.Cts.Cancel();
-                        }
-                        catch (Exception ex)
-                        {
-                            // Suppress exceptions to ensure the task will be awaited.
-                            Log.ErrorOccuredDuringActiveHealthProbingShutdownOnCluster(_logger, cluster.ClusterId, ex);
-                        }
+                Log.StoppedActiveHealthProbingOnCluster(_logger, cluster.ClusterId);
+            }
+        }
 
-                        var response = await probeTask.Task;
-                        response.Dispose();
-                    }
-                    catch (Exception ex)
-                    {
-                        // Suppress exceptions to ensure all responses get a chance to be disposed.
-                        Log.ErrorOccuredDuringActiveHealthProbingShutdownOnCluster(_logger, cluster.ClusterId, ex);
-                    }
-                    finally
-                    {
-                        // Dispose CancellationTokenSource even if the response task threw an exception.
-                        // Dispose() is not expected to throw here.
-                        probeTask.Cts.Dispose();
-                    }
+        private async Task<DestinationProbingResult> ProbeDestinationAsync(ClusterState cluster, DestinationState destination, TimeSpan timeout)
+        {
+            var cts = new CancellationTokenSource(timeout);
+            var waitingForSendAsync = false;
+            try
+            {
+                var request = _probingRequestFactory.CreateRequest(cluster.Model, destination.Model);
+                Log.SendingHealthProbeToEndpointOfDestination(_logger, request.RequestUri, destination.DestinationId, cluster.ClusterId);
+
+                var responseTask = cluster.Model.HttpClient.SendAsync(request, cts.Token);
+                waitingForSendAsync = true;
```

Comment:
Swapping these so Log.ActiveHealthProbeConstructionFailedOnCluster only covers exceptions from _probingRequestFactory.CreateRequest.
```suggestion
                waitingForSendAsync = true;
                var responseTask = cluster.Model.HttpClient.SendAsync(request, cts.Token);
```